tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/wizard/template/CDIWebProjectWizardTemplate.java

```diff
@@ -104,25 +69,29 @@ public void createCDIProjectWithoutBeansXml(){
 		CDIProjectWizard cw = new CDIProjectWizard();
 		cw.open();
 		WebProjectFirstPage fp = new WebProjectFirstPage(cw);
-		fp.setProjectName(PROJECT_NAME);
+		fp.setProjectName(PROJECT_NAME + "2");
 		if (CDIVersion.equals("2.0")) {
 			new DefaultCombo(2).setSelection("Dynamic Web Project with CDI 2.0 (Contexts and Dependency Injection)");
 		}
 		assertEquals(sr.getRuntimeName(),fp.getTargetRuntime());
 		assertEquals("Dynamic Web Project with CDI "+CDIVersion+" (Contexts and Dependency Injection)",fp.getConfiguration());
-		fp.activateFacet("1.8", "Java");
+		
+		activateFacets(fp);
 		cw.next();
 		cw.next();
 		cw.next();
 		CDIInstallWizardPage ip = new CDIInstallWizardPage(cw);
 		ip.toggleCreateBeansXml(false);
 		cw.finish();
-		isCDISupportEnabled(PROJECT_NAME);
-		isCDIFacetEnabled(PROJECT_NAME, CDIVersion);
+		new WaitUntil(new JobIsRunning(), TimePeriod.DEFAULT, false);
+		new WaitWhile(new JobIsRunning(), TimePeriod.LONG);
```

Comment:
another suspicious wait....maybe you can use overloaded finish() to get rid of both waits